index.js

```diff
@@ -81,16 +81,15 @@ function shortenRepoUrl(href, currentUrl = 'https://github.com') {
 	const currentRepo = currentUrl.pathname.slice(1).split('/', 2).join('/');
 
 	/**
-	 * Parse URL
+	 * Parse URL manually to avoid URL encoding and punycode
 	 */
+	const origin = href.split('/', 3).join('/');
+	const pathname = href.slice(origin.length).replace(/[?#].*/, '');
+	const hash = /#.+$/.exec(href)?.[0] ?? '';
+
+	// Use URL exclusively for search parameters because they're too hard to parse
 	const url = new URL(href);
-	const {
-		origin,
-		pathname,
-		search,
-		searchParams,
-		hash,
-	} = url;
+	const {search, searchParams} = url;
```

Comment:
This part, together with the change in `applyToLink`, should resolve the problem at the root: `shortenRepoUrl()` now works directly on what the user sees, not the result of `new URL`

I'll release this as a breaking version